web/unit_tests/cases/attachmentAPI.js

```diff
@@ -0,0 +1,215 @@
+		return editable.id;
+	}
+
+	DynamicElements.assertAttached = function(ele, done) {
+		window.setTimeout(function() {
+			assert.isTrue(keyman.isAttached(ele), "Element tag '" + ele.tagName + "', id '" + ele.id + "' was not attached!");
+
+			if(done) {
+				done();
+			}
+		}, 50);
+	}
+}
+
+Modernizr.on('touchevents', function(result) {
+	if(result) {
+		describe('Attachment (Touch, \'auto\')', function() {
+
+			this.timeout(5000);
+
+			before(function(done) {
+				this.timeout(10000);
+
+				fixture.setBase('unit_tests/fixtures');
+				setupKMW({ attachType:'auto' }, done, 10000);
+			});
+			
+			beforeEach(function() {
+				fixture.load("robustAttachment.html");
+			});
+			
+			after(function() {
+				teardownKMW();
+			});
+			
+			afterEach(function(done) {
+				fixture.cleanup();
+				window.setTimeout(function(){
+					done();
+				}, 500);
+			})
+			
+			describe('Element Type', function() {
+				it('<input>', function(done) {
+					var ID = DynamicElements.addInput();
+					var ele = document.getElementById(ID);
+
+					DynamicElements.assertAttached(ele, done);
+				});
+
+				it('<textarea>', function(done) {
+					var ID = DynamicElements.addText();
+					var ele = document.getElementById(ID);
+
+					DynamicElements.assertAttached(ele, done);
+				});
+
+				it('<iframe>', function(done) {
+					var ID = DynamicElements.addIFrame(function() {
+						var ele = document.getElementById(ID);
+						var innerEle = ele.contentDocument.getElementById('iframe_input');
+
+						assert.isFalse(keyman.isAttached(ele));
+						assert.isNotNull(innerEle);
+						assert.isFalse(keyman.isAttached(innerEle));
+
+						window.setTimeout(function() {
+							done();
+						}, 50);
+					});
+				});
+
+				it('contentEditable=true', function(done) {
+					var ID = DynamicElements.addEditable();
+					var ele = document.getElementById(ID);
+
+					assert.isFalse(keyman.isAttached(ele));
+					done();
+				});
+			});
+		});
+	} else {
+		describe('Attachment (Desktop, \'auto\')', function() {
+
+			this.timeout(5000);
+
+			before(function(done) {
+				this.timeout(10000);
+
+				fixture.setBase('unit_tests/fixtures');
+				setupKMW({ attachType:'auto' }, done, 10000);
+			});
+			
+			beforeEach(function() {
+				fixture.load("robustAttachment.html");
+			});
+			
+			after(function() {
+				teardownKMW();
+			});
+			
+			afterEach(function(done) {
+				fixture.cleanup();
+				window.setTimeout(function(){
+					done();
+				}, 500);
+			})
+			
+			describe('Element Type', function() {
+				it('<input>', function(done) {
+					var ID = DynamicElements.addInput();
+					var ele = document.getElementById(ID);
+
+					DynamicElements.assertAttached(ele, done);
+				});
+
+				it('<textarea>', function(done) {
+					var ID = DynamicElements.addText();
+					var ele = document.getElementById(ID);
+
+					DynamicElements.assertAttached(ele, done);
+				});
+
+				// We still have weird collateral issues with IE and Edge if this is enabled.
```

Comment:
Can you elaborate?

Reply:
Leftover comment from doing development; I disabled the following test initially when having those cross-frame script errors that led to the development of the new `keyman.shutdown` function.